packages/cfd/src/Containers/trade-modal.tsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import { Text, Icon, Money } from '@deriv/components';
+import { TTradingPlatformAccounts, TCFDDashboardContainer, TCFDsPlatformType } from '../Components/props.types';
+import { DetailsOfEachMT5Loginid } from '@deriv/api-types';
+import { CFD_PLATFORMS, getCFDAccountKey, isMobile, capitalizeFirstLetter, mobileOSDetect } from '@deriv/shared';
+import { localize, Localize } from '@deriv/translations';
+import { getPlatformQRCode, mobileDownloadLink, PlatformsDesktopDownload } from '../Helpers/config';
+import { platformsText } from '../Helpers/constants';
+import SpecBox from 'Components/specbox';
+import PasswordBox from 'Components/passwordbox';
+
+type TTradeModalProps = {
+    mt5_trade_account: Required<DetailsOfEachMT5Loginid>;
+    is_eu_user: boolean;
+    onPasswordManager: (
+        arg1: string | undefined,
+        arg2: string,
+        arg3: string,
+        arg4: string,
+        arg5: string | undefined
+    ) => void;
+    toggleModal: () => void;
+    dxtrade_tokens: TCFDDashboardContainer['dxtrade_tokens'];
+    is_demo: string;
+    platform: TCFDsPlatformType;
+};
+
+const getTitle = (market_type: string, is_eu_user: boolean) => {
+    if (is_eu_user) localize('MT5 CFDs');
+    return market_type;
+};
+
+const PlatformIconsAndDescriptions = (platform: TCFDsPlatformType, is_demo: string) => {
+    return (
+        <React.Fragment>
+            <Icon
+                icon={platform === 'derivez' ? 'IcBrandDerivEz' : `IcBrand${capitalizeFirstLetter(platform)}`}
+                size={24}
+            />
+            <div className='cfd-trade-modal__desc'>
+                <Text size='xs' line_height='l' className='cfd-trade-modal__desc-heading'>
+                    <Localize
+                        i18n_default_text='Deriv {{platform}} {{is_demo}}'
+                        values={{
+                            platform: platformsText(platform),
+                            is_demo: is_demo ? 'Demo' : '',
+                        }}
+                    />
+                </Text>
+            </div>
+        </React.Fragment>
+    );
+};
```

Comment:
This can be extracted to become a separate component.
As this TradeModal component is already too long.